tests/eval/test_2fa_duo.py

```diff
@@ -0,0 +1,408 @@
+
+        fresh_browser.switch_to.frame(fresh_browser.find_element_by_id('duo_iframe'))
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        passcode = secrets.test_accounts.duo_code
+        fresh_browser.execute_script("document.getElementsByClassName('passcode-input')[0].value=arguments[0]", passcode)
+        fresh_browser.snap()
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        fresh_browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
+        fresh_browser.close()
+
+
+        # 2FA-8 part b.
+        # Multiple CRNs: IdP displays a CRN chooser and then uses the CRN that was selected.
+        # IdP should display a CRN chooser and then uses the CRN that was selected.
+
+
+def test_remember_me_cookie(utils, sp_url, sp_domain, secrets, netid3, new_tab):
+    """
+    2FA-9 Remember me cookie
+    """
+    chrome_options = webdriver.ChromeOptions()
+    chrome_options.add_experimental_option("prefs", {"profile.default_content_settings.cookies": 1})
+    fresh_browser = Chrome(options=chrome_options)
+
+    sp = ServiceProviderInstance.diafine6
+    with utils.using_test_sp(sp):
+        fresh_browser.get(f'{sp_url(sp)}/shibevalmfa')
+        password = secrets.test_accounts.password
+        fresh_browser.send_inputs(netid3, password)
+        fresh_browser.click(Locators.submit_button)
+        fresh_browser.wait_for_tag('p', 'Use your 2FA device.')
+        fresh_browser.find_element_by_name('rememberme').click()
+        fresh_browser.switch_to.frame(fresh_browser.find_element_by_id('duo_iframe'))
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        passcode = secrets.test_accounts.duo_code
+        fresh_browser.execute_script("document.getElementsByClassName('passcode-input')[0].value=arguments[0]", passcode)
+        fresh_browser.snap()
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        fresh_browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
+        # go to an idp site to retrieve the rememberme cookie
+        fresh_browser.get('https://idp-eval.u.washington.edu/idp')
+        fresh_browser.wait_for_tag('h1', 'not found')
+
+        # look for these cookies:
+        # shib_idp_session, shib_idp_session_ss, and uw-rememberme-sptest03
+        cookie_list = fresh_browser.get_cookies()
+        wanted_cookie_names = ['shib_idp_session', 'shib_idp_session_ss', 'uw-rememberme-sptest03']
+
+        found_cookies = {}
+        for cookie in cookie_list:
+            if cookie['name'] in wanted_cookie_names:
+                found_cookies[cookie['name']] = 1
+
+        assert(len(found_cookies) == 3)
+
+        # b Sign out of the IdP to clear the 2FA session
+        fresh_browser.get('https://diafine6.sandbox.iam.s.uw.edu/Shibboleth.sso/Logout?return=https://idp-eval.u'
```

Comment:
diafine6 will likely be rebuilt and may need a new name.  I'd suggest setting test_service_provider somewhere and referencing that name instead here (and everywhere else).

Reply:
I'm not sure what make diafine6 different from other test SPs but you might describe the host as <something>_service_provider.  Get descriptive with a variable name.